src/test/java/com/iexec/core/replicate/ReplicateSupplyServiceTests.java

```diff
@@ -233,24 +234,22 @@ void shouldNotGetAnyReplicateSinceWorkerAlreadyParticipated() {
         runningTask.setContributionDeadline(DateTimeUtils.addMinutesToDate(new Date(), 60));
         runningTask.setEnclaveChallenge(BytesUtils.EMPTY_ADDRESS);
 
-        final ReplicatesList replicatesList = new ReplicatesList(
+        final ReplicatesList replicatesList = Mockito.spy(new ReplicatesList(
                 CHAIN_TASK_ID,
                 Collections.singletonList(new Replicate(WALLET_WORKER_1, CHAIN_TASK_ID))
-        );
+        ));
 
         workerCanWorkAndHasGas(WALLET_WORKER_1);
         when(taskService.getPrioritizedInitializedOrRunningTask(true, Collections.emptyList()))
                 .thenReturn(Optional.of(runningTask));
         when(workerService.getWorker(WALLET_WORKER_1)).thenReturn(Optional.of(existingWorker));
         when(replicatesService.getReplicatesList(CHAIN_TASK_ID)).thenReturn(Optional.of(replicatesList));
 
-        try (MockedStatic<ReplicatesHelper> replicatesHelper = Mockito.mockStatic(ReplicatesHelper.class)) {
-            replicatesHelper.when(() -> ReplicatesHelper.hasWorkerAlreadyParticipated(replicatesList, WALLET_WORKER_1)).thenReturn(true);
-            Optional<WorkerpoolAuthorization> oAuthorization =
-                    replicateSupplyService.getAuthOfAvailableReplicate(workerLastBlock, WALLET_WORKER_1);
+        when(replicatesList.hasWorkerAlreadyParticipated(WALLET_WORKER_1)).thenReturn(true);
```

Comment:
Are you sure `when(...).then(...)` works on `Spy` objects? I think we should use `do(...).when(...)` instead.

Reply:
Good, fair enough. I'm pretty sure I had a different behavior before but maybe it wasn't the same thing or new versions changed it. Thanks for verifying that.